test/spec/util-spec.js

```diff
@@ -53,4 +53,25 @@ describe("util", function () {
       expect(actual).toEqual(expected);
     });
   });
+  describe("detectIntersection",function () {
+    let onIntersect;
+    beforeEach(function() {
+      onIntersect = jasmine.createSpy("onIntersect");
+      jasmine.clock().install();
+    });
+
+    it("should call onIntersect", function () {
+      const {detectIntersection} = cloudinary.Util;
+
+      setTimeout(function() {
+        detectIntersection({}, onIntersect);
+      }, 100);
+
+      expect(onIntersect).not.toHaveBeenCalled();
```

Comment:
Why do we need the timer here?
Can't you call expect before and after you run detectIntersection?

Reply:
Simplify the test and state that it checks worst case scenario and fallsback to call the callback